clippy_lints/src/manual_non_exhaustive.rs

```diff
@@ -123,44 +125,48 @@ fn check_manual_non_exhaustive_struct(cx: &EarlyContext<'_>, item: &Item, data:
         matches!(field.vis.node, VisibilityKind::Inherited)
     }
 
-    fn is_non_exhaustive_marker(name: &Option<Ident>) -> bool {
-        name.map(|n| n.as_str().starts_with('_')).unwrap_or(true)
+    fn is_non_exhaustive_marker(field: &StructField) -> bool {
+        is_private(field) && field.ty.kind.is_unit() && field.ident.map_or(true, |n| n.as_str().starts_with('_'))
+    }
+
+    fn find_header_span(cx: &EarlyContext<'_>, item: &Item, data: &VariantData) -> Span {
+        let delimiter = match data {
+            VariantData::Struct(..) => '{',
+            VariantData::Tuple(..) => '(',
+            _ => unreachable!("`VariantData::Unit` is already handled above"),
```

Comment:
```suggestion
            VariantData::Unit => unreachable!("`VariantData::Unit` is already handled above"),
```

(This would be a nice pedantic lint. If only one variant is left and matched with `_`, replace it with the variant)

Reply:
Good catch! I'm going to open the issues for the two lints we discussed about in this PR.

Reply:
FTR, [here](https://github.com/rust-lang/rust-clippy/issues/5556) is the issue